### ISSUE TITLE
Integration suddenly not wokring

### ISSUE THREAD

**Issue by t112013** [fondberg/easee_hass#315]
### The problem

Connected my car and then i got status unkown7 isted of connected 
<img width="327" alt="Skjermbilde 2023-04-28 kl  22 41 41" src="https://user-images.githubusercontent.com/55782238/235250659-9c61bce9-53f3-44fb-a761-83d57c5f07d7.png">

So i reloaded the integration and now nothing seems to make sence.
The easee app works just fine
<img width="1015" alt="Skjermbilde 2023-04-28 kl  22 41 26" src="https://user-images.githubusercontent.com/55782238/235250791-da1ee7ee-c6ac-4b22-8f3a-2b6759e91551.png">

I have two chargers, both of them show the same disconnected values. Also tried rebooting the home assistant host system just in case.

Seems to be an authorization issue ?
I have deleted and re-added the integration but no luck.

### Version of Easee integration having the issue?

0.9.50

### Version of Home Assistant Core having the issue?

2023.4.6

### Anything in the logs that might be useful for us?

2023-04-28 22:54:58.593 DEBUG (MainThread) [pyeasee.easee] Unautorized (401:  https://api.easee.com/firmware/EHYUJEXW/latest)
2023-04-28 22:54:58.593 DEBUG (MainThread) [pyeasee.easee] Re authorizing due to 401
2023-04-28 22:54:58.593 DEBUG (MainThread) [pyeasee.easee] getting token for user: xxxx@xxxxx.xxx
2023-04-28 22:54:58.663 DEBUG (MainThread) [pyeasee.easee] TOKEN: {'accessToken': 'xxxxxxxxxxxxxx', 'expiresIn': 86400, 'accessClaims': ['User'], 'tokenType': 'Bearer', 'refreshToken': 'xxxxxxxxxxxxx'}
2023-04-28 22:54:58.663 DEBUG (MainThread) [pyeasee.easee] Unautorized (401:  https://api.easee.com/firmware/EH66B2Q4/latest)

**Comment by LauWs** [comment]
I have the exact same issue.
Tried removing it and then install again, no luck.

**Comment by pvprodk** [comment]
Same problem

**Comment by t112013** [comment]
BTW in case it may help i read in the newspapers last week that easee had stopped paying their providers, i imagine some azure or aws just stoopped some api services ?  just an idea hehe

**Comment by t112013** [comment]
maybe some url change. 
https://api.easee.com/  it gives forbidden. 
The dev page uses .cloud 
<img width="677" alt="Skjermbilde 2023-04-29 kl  00 27 57" src="https://user-images.githubusercontent.com/55782238/235264217-85b04d06-4134-4230-ab4f-5a44ffbdda87.png">

Maybe that is the issue ?

**Comment by olalid** [comment]
I am seeing this too (but only after a restart). I am investigating...

**Comment by olalid** [comment]
A quick and dirty workaround in https://github.com/fondberg/easee_hass/pull/316 should solve this, at least temporarily until we understand what is going on.

**Comment by olalid** [comment]
Please try latest version (0.9.51) that was just released.

**Comment by pvprodk** [comment]
> Please try latest version (0.9.51) that was just released.

Solved the problem here 👌🏼☺️

**Comment by PeteHassio** [comment]
Thanks for the quick fix! The status "unknown_7" is still there when the car is connected.

**Comment by astrandb** [comment]
Status 7 is a new (or previously not used) status code. We will try fo find what it means. Perhaps you can see what the app displays?

**Comment by PeteHassio** [comment]
The app shows "Waiting for approval". After the initial approval, the status in home assistant is displayed correctly again. Thanks for the help!

**Comment by astrandb** [comment]
OK, we will try to get the formal explanation from Easee. We will fix a proper message in next release. At least we know what` Unknown 7` is.

**Comment by t112013** [comment]
It must be something else going on here becasue i have two chargers, one of them gives me unknown 7 and the other one gives me Awaiting start, while the app was showing waiting for approval. ( i have to approve the chargers every time i connect them, using the start command)
My automations have wowrked flawlessly until yesterday.
The only thing i can see is that they dont have the same software version even i bough them together so i guess they are exacly the same hardware.

**Comment by astrandb** [comment]
There are a number of different issues that have surfaced the last few days due to changes in the Easee public API. 
- A "new" status message `unknown_7` has been displayed. The mapping to clear text is fixed in the integration code and appear in next release.
- The question why `status 7` / `awaiting_authorization` is reported by the  must be put to Easee. It has nothing to do with this integration.
- A new API endpoint for firmware version has been introduced  and the old is deprecated. This integration had an imperfect error handling when this new endpoint was failing. This is fixed in latest release. The API endpoint still fails but now it only affects the firmware sensor in this integration. 
- The URL-link on the device page has changed. Fixed in next released of the integration.

We try fix the issues as soon as we understand the underlying causes.

Please create new issues for each problem. When a new issue is added to this thread it is difficult for users (and developers) to see if a problem is already reported and if a fix is underway.

**Comment by t112013** [comment]
Oh thanks for the great feedback. When are you planning on releasing the new version. Or is ir on the beta channel? 
Cheers

**Comment by olalid** [comment]
@Epenance have you read this article about the difference between start/stop/pause/resume?
https://developer.easee.cloud/docs/settings
I.e start is not a command used to start charging, it is to initiate authorisation.
You might what to try using resume instead.

**Comment by Epenance** [comment]
@olalid Thanks for the hint, I will try to do a test today when I get home using Resume, but I still assume that I would need to use start initially since I have authentication required on my charger.

**Comment by Epenance** [comment]
Using Resume made no difference, however I'll leave a support ticket at Easee to get more information rather than hijacking this issue :)

**Comment by olalid** [comment]
Ok, I assume it has to be something that has changed on Easees side then.

**Comment by t112013** [comment]
i can confirm that the firmware version v310 makes  the difference, until yesterday or so  this charger was not displaying unkown 7 but my other charger did (they didnt have version v310, now they both do)

**Comment by olalid** [comment]
Ok, thanks for letting us know.

**Comment by olalid** [comment]
Please test the release, hopefully it solves the status code issue.

**Comment by SlayerOwl** [comment]
> Please test the latest release, hopefully it solves the status code issue.

I'm running v0.9.52 and all my states are 'Unavailable'.

![image](https://github.com/fondberg/easee_hass/assets/47760225/6208ae03-fee4-40cc-8dd9-667e712cfb34)

**Comment by iboboc** [comment]
My integration is partially working after last firmware update.
Easee home, firmware v310, integration 0.9.52.

It shows only partial data, the most important ones are unavailable.

![image](https://github.com/fondberg/easee_hass/assets/2733631/06a3288e-8214-4b4c-b6e0-bc211a8290fe)

Any idea or sugesstion?
Thanks.

**Comment by olalid** [comment]
Are you sure the entities are enabled?
Any error in the log file?

**Comment by iboboc** [comment]
Damn' it, sorry about that.

I never disabled it, so being disabled, my assumption was that they are not working/receiving data.
Even after removing the integration and reinstall it, still those entities were not available.

I re-enabled each one manually and it is working. Sorry for wasting your time and help for feedback.

**Comment by olalid** [comment]
Strange that they were disabled though, this is not under the integrations control really, but it should not randomly change.
In any case, no problem.

**Comment by toredash** [comment]
I think this is related to when the charger (api/chargers/{id}) is returning `[{"levelOfAccess": 2}]`, which is when the access mode is set to "Key only (Public)". I've done via the App, but I'm unable to find this API call in their docs.

In their docs they state that code 7 is: `AwaitingAuthentication = 7`, but there is also a 8: `Deauthenticating = 8`

Source: https://developer.easee.cloud/docs/enumerations (under OpMode).

AFAIK, this isn't supported in the python lib by @fondberg 

Does this provide any help @olalid  ?

**Comment by olalid** [comment]
A "translation" of status code 7 was added in the most recent release v0.9.52, so as far as I am concerned, the original problems that was reported here was solved a few weeks ago.
We have just kept this issue open waiting for feedback from the original reporter.
For some reason people seem to think that this thread is a good place to report other unrelated problems, so I am closing it now to remove that temptation for everyone :)

I am not sure what you mean about "isn't supported in the python lib". What are you missing exactly?

Thanks for the pointer to where to find official documentation for what 7 (and 8) means.
